### PR TITLE
AzureRT PR

### PR DESCRIPTION
**Pull request by hyonholee**
- Combine Get-AzureVMImage and Get-AzureVMImageDetail
- Combine Get-AzureVMExtensionImage and Get-AzureVMExtensionImageDetail
- Fix piping issue of Remove-AzureVMDataDisk and Remove-AzureVMNetworkInterface
- Update the ouput of Remove-AzureAvailabilitySet
- Add Force parameter for Remove-AzureAvailabilitySet

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hyonholee__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (hylee). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+// 
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// 
+
+using System.Net;
+
+namespace Microsoft.Azure.Commands.Compute.Models
+{
+    public class PSOperationResponse
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs]: It's duplicate, and please use the following one:
```
    public class PSOperation
    {
        public string RequestId { get; set; }

        public HttpStatusCode StatusCode { get; set; }
    }
```

File: src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs [hunk 2]
```diff
@@ -33,6 +33,7 @@ public string ResourceGroupName
         {
             get
             {
+                if (string.IsNullOrEmpty(Id)) return null;
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs]: This is a generated model class code, and it shall not be changed manually. It's better to update the script.

File: src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs [hunk 3]
```diff
@@ -134,5 +135,37 @@ public string StorageProfileText
         {
             get { return JsonConvert.SerializeObject(StorageProfile, Formatting.Indented); }
         }
+
+        [JsonIgnore]
+        public string [] DataDiskNames
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs]: same

File: src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs [hunk 4]
```diff
@@ -134,5 +135,37 @@ public string StorageProfileText
...
+                foreach (var disk in StorageProfile.DataDisks)
+                {
+                    dataDiskNames.Add(disk.Name);
+                }
+                return dataDiskNames.ToArray();
+            }
+        }
+
+        [JsonIgnore]
+        public string[] NicIds
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs]: same. BTW, better to use full name like NetworkInterfaceIds or ..Names

**Comment by huangpf** [comment]
I have concerns about ```Fix piping issue of Remove-AzureVMDataDisk and Remove-AzureVMNetworkInterface```, and I suggest we don't patch this in this sprint. Other than that LGTM.

**Comment by markcowl** [comment]
@huangpf, is this ready to be considered for merge?

**Comment by markcowl** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs (hunk 1 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs]: There are a bunch of child class of PSOperation, so if you use PSOperation and format it, all other cmdlets are affected, and show a wrong format. That is why I created a new class.

File: src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs (hunk 2 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs]: Yes, but this is a fix for a bug I found (Not relavant to this PR, though).  We could update the script later.

Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSVirtualMachine.cs]: I would rather not; please try to update the script first.

File: src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/RemoveAzureVMDataDiskCommand.cs [hunk 5]
```diff
@@ -38,13 +38,14 @@ public class RemoveAzureVMDataDiskCommand : AzurePSCmdlet
...
 
+        [Alias("Name")]
         [Parameter(
             Mandatory = true,
             Position = 1,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = HelpMessages.VMDataDiskName)]
         [ValidateNotNullOrEmpty]
-        public string Name { get; set; }
+        public string [] DataDiskNames { get; set; }
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Config/RemoveAzureVMDataDiskCommand.cs]: ```string[]```

File: src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs (hunk 1 above)
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Models/PSOperationResponse.cs]: May I know what's the specific wrong format about? If something is wrong, it's better to fix it correctly, isn't it?

**Pull request merged**